runelite-client/src/main/java/net/runelite/client/plugins/playerscouter/PlayerScouter.java

```diff
@@ -0,0 +1,334 @@
+	}
+
+	private void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() == GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		blacklist.clear();
+	}
+
+	private void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals("playerscouter"))
+		{
+			return;
+		}
+		updateConfig();
+	}
+
+	private void onInteractingChanged(InteractingChanged event)
+	{
+		if ((event.getSource() instanceof Player) && (event.getTarget() instanceof Player))
+		{
+			final Player source = (Player) event.getSource();
+			final Player target = (Player) event.getTarget();
+
+			if (source == client.getLocalPlayer())
+			{
+				if (!PvPUtil.isAttackable(client, target))
+				{
+					return;
+				}
+
+				playerContainer.forEach(player ->
+				{
+					if (player.getPlayer() == target)
+					{
+						player.setTimer(16);
+						player.setTarget(true);
+					}
+				});
+			}
+			else if (target == client.getLocalPlayer())
+			{
+				if (!PvPUtil.isAttackable(client, source))
+				{
+					return;
+				}
+				playerContainer.forEach(player ->
+				{
+					if (player.getPlayer() == source)
+					{
+						player.setTimer(16);
+						player.setTarget(true);
+					}
+				});
+			}
+		}
+	}
+
+	private void onAnimationChanged(AnimationChanged event)
+	{
+		final Actor actor = event.getActor();
+
+		if (actor.getInteracting() != client.getLocalPlayer())
+		{
+			return;
+		}
+
+		if (!(actor instanceof Player))
+		{
+			return;
+		}
+
+		if (PvPUtil.isAttackable(client, (Player) actor) && actor.getAnimation() != -1)
+		{
+			playerContainer.forEach(player ->
+			{
+				if (player.getPlayer() == actor)
+				{
+					player.setTimer(16);
+					player.setTarget(true);
+					player.setAttacking(true);
+				}
+			});
+		}
+	}
+
+	private void onPlayerSpawned(PlayerSpawned event)
+	{
+		final Player player = event.getPlayer();
+
+		if (player == client.getLocalPlayer())
+		{
+			return;
+		}
+
+		if (!blacklist.isEmpty() && blacklist.keySet().contains(player.getName()))
+		{
+			return;
+		}
+
+		executorService.submit(() ->
+		{
+			SingleHiscoreSkillResult result;
+
+			try
+			{
+				result = HISCORE_CLIENT.lookup(player.getName(), HiscoreSkill.PRAYER);
+			}
+			catch (IOException ex)
+			{
+				log.warn("Error fetching Hiscore data " + ex.getMessage());
+				return;
+			}
+
+			playerContainer.add(new PlayerContainer(player, result.getSkill()));
+			blacklist.put(player.getName(), client.getTickCount() + this.timeout);
+		});
+	}
+
+	private void onPlayerDespawned(PlayerDespawned event)
+	{
+		final Player player = event.getPlayer();
+
+		playerContainer.removeIf(p -> p.getPlayer() == player);
+	}
+
+	private void onGameTick(GameTick event)
+	{
+		resetBlacklist();
+
+		if (playerContainer.isEmpty())
+		{
+			return;
+		}
+
+		playerContainer.forEach(player ->
+		{
+			Utils.reset(player);
+			Utils.update(player, itemManager, 6, WILD_LOCS);
+
+			if (player.getRisk() > this.minimumRisk && checkWildy())
```

Comment:
Seems like this will be scanning player values for all players even if you aren't in wildy -- perf concern